spec/system/purchase_system_spec.rb

```diff
@@ -216,6 +216,13 @@
           visit edit_purchase_path(@user.organization.short_name, purchase)
           expect(page).to have_content("Still haven't found what you're looking for")
         end
+
+        it "The storage location is correct" do
+          storage1 = create(:storage_location, name: "storage1")
+          purchase = create(:purchase, storage_location: storage1)
+          visit edit_purchase_path(@organization.to_param, purchase)
+          assert page.has_select? "Storage Location", selected: "storage1"
+        end
```

Comment:
Agree with @seanmarcia (his comment has been abandoned since you force pushed the branch - please don't do that with open PRs that have comments!)

You can inspect the rendered response in a request test. If you're not interacting with it in any way, that should be fine. System tests should be limited to interactivity and JavaScript testing since it's slower and more brittle.

Reply:
Can another test be added to ensure the `new` behavior is also as expected?

Reply:
I've moved the test into requests, and added a new test to ensure the `new` behavior. Let me know if these address the changes you wanted.